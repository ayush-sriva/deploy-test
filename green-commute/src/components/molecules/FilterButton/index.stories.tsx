import { ComponentStory, ComponentMeta } from '@storybook/react';
import FilterButton from '.';
import Icon from '../../atoms/Icon';

export default {
  title: 'components/molecules/FilterButton',
  component: FilterButton,
} as ComponentMeta<typeof FilterButton>;

const Template: ComponentStory<typeof FilterButton> = (args) => <FilterButton {...args} />;

export const FilterButtonDemo = Template.bind({});

FilterButtonDemo.args = {
  startIcon: <Icon url={'/assets/icons/filterIcon.png'} />,
  children: 'Filter',
};
