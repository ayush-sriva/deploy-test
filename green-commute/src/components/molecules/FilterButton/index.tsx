/**
 * @author Bhavani Somanchi <bhavani.somanchi@zemosolabs.com>
 */

import { Button, styled } from '@mui/material';
import theme from '../../../theme';

interface IconButtonProps {
  startIcon?: React.ReactNode;
  children?: React.ReactNode;
  onClickHandler?: React.MouseEventHandler<HTMLButtonElement>;
}

const StyledButton = styled(Button)({
  lineHeight: '22px',
  color: theme.palette.secondary.dark,
  width: '137px !important',
  height: '56px',
  backgroundColor: theme.palette.primary.contrastText,
  borderRadius: '32px',
  '& .MuiButton-startIcon': {
    marginRight: '16px',
  },
  marginLeft: '12px',
  marginTop: '2.3rem',
});

const IconButtonComponent: React.FC<IconButtonProps> = ({ startIcon, children, onClickHandler }) => {
  return (
    <StyledButton startIcon={startIcon} onClick={onClickHandler}>
      {children}
    </StyledButton>
  );
};

export default IconButtonComponent;
