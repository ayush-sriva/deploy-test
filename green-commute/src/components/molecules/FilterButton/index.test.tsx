import WorkOutline from '@mui/icons-material/WorkOutline';
import renderer from 'react-test-renderer';
import FilterButton from '.';

describe('Filter Button', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<FilterButton startIcon={<WorkOutline />} children='Filter' />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
