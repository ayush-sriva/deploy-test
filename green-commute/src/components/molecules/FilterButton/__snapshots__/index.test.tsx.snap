// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Button renders correctly 1`] = `
<button
  className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-a5aw58-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="WorkOutlineIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M14 6V4h-4v2h4zM4 8v11h16V8H4zm16-2c1.11 0 2 .89 2 2v11c0 1.11-.89 2-2 2H4c-1.11 0-2-.89-2-2l.01-11c0-1.11.88-2 1.99-2h4V4c0-1.11.89-2 2-2h4c1.11 0 2 .89 2 2v2h4z"
        fillRule="evenodd"
      />
    </svg>
  </span>
  Filter
</button>
`;
